### ISSUE TITLE
Custom theme is lost after switching dark mode

### ISSUE THREAD

**Issue by IlyaSemenov** [react-native-elements/react-native-elements#2647]
**Explain what you did (Required)**

I am running an app that updates the theme dynamically  **and** uses custom themes, with this:

```tsx

const themes = {
  light: { colors: { primary: "violet" } },
  dark: { colors: { primary: "red" } },
}

export const MyApp = () => {
  const scheme = useColorScheme() // "light" | "dark"
  return (
    <ThemeProvider useDark={scheme === "dark"} theme={themes[scheme]}>
      <MainView />
    </ThemeProvider>
  )
}
```

I start the app in Dark mode on iOS simulator, then press Cmd-Shift-A twice (switch to light and back to dark).

**Expected behavior (Required)**

The app shows custom dark theme (red primary color), then switches to custom light theme (violet primary color), then switches to custom dark theme (red primary color).

**Describe the bug (Required)**

The app shows custom dark theme (red primary color), then switches to **built-in** light theme (blue primary color), then switches to **built-in** dark theme (blue primary color).

In other words, after first dark mode switch, all custom theming is lost.

**To Reproduce (Required)**

I understand why do you mark this as required, but in this case the bug is clearly obvious. It was introduced recently in #2572.

This code is obviously broken: https://github.com/react-native-elements/react-native-elements/blob/0c7dcda33a005ce7c47169f5f0e6e2b978c74f75/src/config/ThemeProvider.js#L34-L40

it overwrites the current theme with the default colors - exactly the observed behaviour.

If you absolutely insist, I will create an expo demo, but again, the code is clearly broken.

**Your Environment (Required):**

| software              | version |
| --------------------- | ------- |
| react-native-elements | 3.0.0-alpha.1       |
| react-native          |     0.63.2    |

**Comment by vividh** [comment]
Seems like a straightforward fix. May I claim this issue?

**Comment by vividh** [comment]
Didn't hear back, but have fixed this in https://github.com/react-native-elements/react-native-elements/pull/2650. Took the liberty of also adding another test case. Please check it out and let me know if any changes are required. Thanks :)
